api/apps/search/routes.coffee

```diff
@@ -0,0 +1,13 @@
+_ = require 'underscore'
+search = require '../../lib/elasticsearch.coffee'
+queries = require './queries.coffee'
+
+# GET /api/search
+@index = (req, res, next) ->
+  search.client.search(
+      body:
+        query: queries.matchAll(req.query.term)
+      , (error, response) ->
+        return console.log("nothing matched for: #{req.query.term}") if error
+        res.send response.hits
```

Comment:
I assume since you don't specify indexes here, this will hit all the indexes. This could potentially be constrained using the [index parameter](https://www.elastic.co/guide/en/elasticsearch/reference/2.1/multi-index.html).

Reply:
i tried to limit it to specific types, but then the list got long enough, would this be a performance hit?